src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProcessor.cs

```diff
@@ -31,63 +57,133 @@ public ConsoleLoggerProcessor(IConsole console, IConsole errorConsole)
                 Name = "Console logger queue processing thread"
             };
             _outputThread.Start();
+            _meter.CreateObservableGauge<long>("queue-size", GetQueueSize);
         }
 
         public virtual void EnqueueMessage(LogMessageEntry message)
         {
-            if (!_messageQueue.IsAddingCompleted)
-            {
-                try
-                {
-                    _messageQueue.Add(message);
-                    return;
-                }
-                catch (InvalidOperationException) { }
-            }
-
-            // Adding is completed so just log the message
-            try
+            // cannot enqueue when adding is completed
+            if (!Enqueue(message))
             {
                 WriteMessage(message);
             }
-            catch (Exception) { }
         }
 
         // for testing
         internal void WriteMessage(LogMessageEntry entry)
         {
-            IConsole console = entry.LogAsError ? ErrorConsole : Console;
-            console.Write(entry.Message);
+            try
+            {
+                var messagesDropped = Interlocked.Exchange(ref _messagesDropped, 0);
+                if (messagesDropped != 0)
+                {
+                    System.Console.Error.WriteLine($"{messagesDropped} message(s) dropped because of queue size limit. Increase the queue size or decrease logging verbosity to avoid this.{Environment.NewLine}");
+                }
+
+                IConsole console = entry.LogAsError ? ErrorConsole : Console;
+                console.Write(entry.Message);
+            }
+            catch (Exception)
+            {
+                CompleteAdding();
```

Comment:
If `console.Write(entry.Message);` on line 84 throws, then we call `CompleteAdding()` to gracefully stop logging. I believe that is also happening in the original code.

#### comparison with original code:

In the original code, `WriteMessage` is guarded by the try/catch block meant for any exception that might occur due to calling `GetConsumingEnumerable` (refer to [docs](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.blockingcollection-1.getconsumingenumerable?view=net-6.0)).

I believe if the `console.Write(entry.Message)` used to throw before the logger would gracefully stop due to this implementation. Even though it may not have been the intention then.

link to original code snippet:

https://github.com/dotnet/runtime/blob/24f5de4bb10976dfb391e9b6affe5481f55a2d00/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProcessor.cs#L57-L80

Reply:
Looks we used to catch the exception inside ProcessLogQueue before and then CompleteAdding there. I guess the logic is moved now here instead. 

Maybe better to catch the exception in ProcessLogQueue as we did before? We didn't catch the exceptions in WriteMessage before

Reply:
I sent my comment without seeing @maryamariyan comment. 

> I think when console.Write(entry.Message) throws, in the original code, the logger gracefully stops due to this implementation. Even though it may not have been the intention then.

How does the logger stop at that time? could you please clarify with the following example? In the old code when calling `ConsoleLogger.Log` which will call `ConsoleLoggerProcessor.EnqueueMessage` which calls `WriteMessage`. If `WriteMessage` throws, nothing will happen and CompleteAdding will not be called.

Reply:
The use case is described in unit the test called `ThrowDuringProcessLog_ShutsDownGracefully`

Reply:
Just confirmed the behaviour in `ThrowDuringProcessLog_ShutsDownGracefully` is the same before or after the changes in this PR.

Reply:
I am not going to unresolve it but I hope I can get answer to the scenario I mentioned:

`In the old code when calling ConsoleLogger.Log which will call ConsoleLoggerProcessor.EnqueueMessage which calls WriteMessage. If WriteMessage throws, nothing will happen and CompleteAdding will not be called.` 

If the same exact scenario is executed with your change, I expect some different behavior as it will call `CompleteAdding`, I don't know if `ThrowDuringProcessLog_ShutsDownGracefully` is testing this exact scenario. I guess. Am I missing anything?